src/lib/hooks/useSearch.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from "react";
+
+interface DataResponse {
+  list: any[];
+  totalCount: number;
+}
+
+export const useSearch = (
+  data: DataResponse,
+  setCurrentPage: (page: number) => void,
```

Comment:
p4;
useSearch를 호출하여 사용하는 곳을 보면 setCurrentPage에 빈 더미 함수를 전달하는 곳도 있는 것 같아요!
따라서 이 함수는 옵셔널하게 전달할 수 있도록 처리하는게 좋을 것 같아요!

```suggestion
  setCurrentPage?: (page: number) => void,
```